mlir/lib/Gradient/Transforms/BufferizationPatterns.cpp

```diff
@@ -247,6 +247,92 @@ struct BufferizeForwardOp : public OpConversionPattern<ForwardOp> {
     }
 };
 
+struct BufferizeReverseOp : public OpConversionPattern<ReverseOp> {
+    using OpConversionPattern::OpConversionPattern;
+
+    LogicalResult match(ReverseOp op) const override
+    {
+        // Only match here if we have all memref arguments and return values.
```

Comment:
Should check that this is empty

Reply:
https://github.com/PennyLaneAI/catalyst/pull/822/commits/06ccbeee620b7b91365927737dd73f8d49d09349